index.js

```diff
@@ -26,22 +40,31 @@ function attacher(destination, options) {
   return destination ? bridge(destination, settings) : mutate(settings)
 }
 
-// Bridge-mode.
-// Runs the destination with the new mdast tree.
+/**
+ * Bridge-mode.
+ * Runs the destination with the new mdast tree.
+ * @param {FrozenProcessor} destination
+ * @param {Settings} options
+ * @returns {Transformer}
+ */
 function bridge(destination, options) {
   return transformer
-  function transformer(node, file, next) {
-    destination.run(hast2mdast(node, options), file, done)
-    function done(err) {
-      next(err)
-    }
```

Comment:
@wooorm my guess is that, `done` and `next` are used here is to support a Pipeline with an async plugin applied? Is that accurate? :thinking: :detective:

Reply:
Yes, `next` and `done` are needed. Otherwise async plugins will fail

Reply:
Ah, gotcha. Do we need to pass `next(err, node, file)` then? Or is there another way here?  When transformer is a `Transformer`, next is a `(error: Error, tree: Node, file: vfile.VFile) => Record<string, unknown>`. Not passing the args results in:

>Expected 3 arguments, but got 1.

Please advise.

Reply:
That may be a bug in the upstream typing https://github.com/unifiedjs/unified/blob/699b97a5910392b9cae6d15ab7cb279cb3dd71c8/types/index.d.ts#L377-L384
which currently isn't tested https://github.com/unifiedjs/unified/blob/699b97a5910392b9cae6d15ab7cb279cb3dd71c8/types/unified-tests.ts which may explain why this hasn't been caught yet.
Next should only require one parameter https://github.com/unifiedjs/unified#function-nexterr-tree-file

---

`unified`'s main branch is midway through a version 10 upgrade (https://github.com/unifiedjs/unified/issues/121).
To get this change in place with unified version 9, a `// @ts-ignore` may be needed.

Reply:
I see. Well, I tried to address it without ignoring. Please take a look, I can easily revert if it's too much.